test/init_test.go

```diff
@@ -50,7 +51,27 @@ func init() {
 	flag.BoolVar(&skipRootUserTests, "skipRootUserTests", false, "Skip tests that require root user")
 }
 
+// excludeForArchitecture checks if name of calling function(name of the test) is in in the excluded list
+func excludeForArchitecture(callerLevel int) bool {
+	//get the Caller and the func name after
+	pc, _, _, _ := runtime.Caller(callerLevel)
+	funcFullName := runtime.FuncForPC(pc).Name()
+	lastSlash := strings.LastIndexByte(funcFullName, '/')
+	if lastSlash < 0 {
+		lastSlash = 0
+	}
+	lastDot := strings.LastIndexByte(funcFullName[lastSlash:], '.') + lastSlash
+	funcName := funcFullName[lastDot+1:]
+
+	//check if func name, which equal to test name is excluded for special architecture
+	return IsExcluded(funcName)
+}
+
 func setup(t *testing.T, fn ...func(*testing.T, *clients, string)) (*clients, string) {
+	if excludeForArchitecture(1) || excludeForArchitecture(2) {
```

Comment:
nit: the integers here don't have much context. I would find it more clear to move the 1, 2 and || check up into the excludeForArchitecture function itself.

Reply:
not a topic any more, as per your great suggestion this code was removed